### PR TITLE
Use foreman apache configs to unblock Pulp 3

### PR DESCRIPTION
**Pull request by ehelms**
This module had declared it's own vhost using a similar naming convention and directory structure as Foreman's. This causes a conflict when pulpcore is enabled since pulpcore relies on adding fragments to Foreman's vhost declaration specifically around the use of `05-foreman-ssl.d`. To solve this, and bring greater continuity I have attempted to declare the `foreman::config::apache` class and re-use it here.

### CODE REVIEW

File: manifests/apache.pp [hunk 1]
```diff
@@ -1,101 +1,46 @@
...
+    ssl_protocol            => undef,
+    ssl_verify_client       => 'optional',
+  }
 
-  $priority = '05'
+  foreman::config::apache::fragment { 'katello':
+    ssl_content => file('katello/katello-apache-ssl.conf'),
+  }
 
   $proxy_pass_https = [
```
Comment (ehelms) [inline manifests/apache.pp]: @ekohl I wasn't sure how to declare this additional proxy pass information to the class.

**Comment by ehelms** [comment]
@ekohl general thoughts on this overall approach? It is related to this change https://github.com/theforeman/forklift/pull/1105

File: manifests/apache.pp [hunk 2]
```diff
@@ -1,101 +1,46 @@
...
+		server_ssl_port         => $foreman::params::server_ssl_port,
+		ssl_certs_dir           => '',
+		ssl_crl                 => '',
+		user                    => $foreman::params::user,
+		passenger_prestart      => false,
+		passenger_min_instances => 0,
+		passenger_start_timeout => 0,
+		foreman_url             => $foreman::params::foreman_url,
+		ipa_authentication      => false,
+    proxy_backend           => "http://0.0.0.0:${katello_devel::rails_port}/",
```
Comment (ekohl) [inline manifests/apache.pp]:
```suggestion
    proxy_backend           => "http://localhost:${katello_devel::rails_port}/",
```

File: manifests/apache.pp [hunk 3]
```diff
@@ -1,101 +1,46 @@
...
+  include foreman::params
+
+  class { 'foreman::config::apache':
+    passenger               => false,
+    ssl                     => true,
+    ssl_cert                => $certs::apache::apache_cert,
+    ssl_key                 => $certs::apache::apache_key,
+    ssl_ca                  => $certs::apache::ca_cert,
+    ssl_chain               => $certs::apache::ca_cert,
+		app_root                => '/home/vagrant/foreman',
```
Comment (ekohl) [inline manifests/apache.pp]: Looks like tabs instead of spaces

File: manifests/apache.pp [hunk 4]
```diff
@@ -1,101 +1,46 @@
...
+		serveraliases           => $foreman::params::serveraliases,
+		server_port             => $foreman::params::server_port,
+		server_ssl_port         => $foreman::params::server_ssl_port,
+		ssl_certs_dir           => '',
+		ssl_crl                 => '',
+		user                    => $foreman::params::user,
+		passenger_prestart      => false,
+		passenger_min_instances => 0,
+		passenger_start_timeout => 0,
+		foreman_url             => $foreman::params::foreman_url,
```
Comment (ekohl) [inline manifests/apache.pp]:
```suggestion
    foreman_url             => "https://${certs::apache::hostname}/",
```

File: manifests/apache.pp (hunk 1 above)
Comment (ekohl) [reply manifests/apache.pp]: I actually wonder if this is really needed since I'm not sure what listens on port 6006. Looks like the storybook according to 3a61896824f67b050f4ce31467adea1e1f7e5b7c but perhaps the devs should just go to `http://$hostname:6006` for that.

**Comment by ekohl** [comment]
I'd like to see a version that uses https://github.com/theforeman/puppet-foreman/pull/800 because it should be simpler.

File: manifests/apache.pp (hunk 2 above)
Comment (ehelms) [reply manifests/apache.pp]: Developers still access the dev server directly, which means it does not run on localhost.

Comment (ekohl) [reply manifests/apache.pp]: I hope they let it listen on `*` or `0.0.0.0` which means it still listens on localhost. I also wonder why they need to access the dev server directly if Apache is in front of it.

File: manifests/apache.pp [hunk 5]
```diff
@@ -1,100 +1,18 @@
...
-    proxy_pass            => $proxy_pass_https,
-    proxy_preserve_host   => true,
-    request_headers       => ["set X_FORWARDED_PROTO 'https'"],
+  class { 'foreman::config::apache':
+    passenger     => false,
+    ssl           => true,
+    ssl_cert      => $certs::apache::apache_cert,
+    ssl_key       => $certs::apache::apache_key,
+    ssl_ca        => $certs::apache::ca_cert,
+		app_root      => '/home/vagrant/foreman',
```
Comment (ekohl) [inline manifests/apache.pp]: This isn't needed
```suggestion
```

File: manifests/apache.pp [hunk 6]
```diff
@@ -1,100 +1,18 @@
...
+    proxy_backend => "http://0.0.0.0:${katello_devel::rails_port}/",
```
Comment (ekohl) [inline manifests/apache.pp]: I still think this can just be localhost

File: manifests/apache.pp [hunk 7]
```diff
@@ -1,100 +1,18 @@
...
-    ssl_proxyengine       => true,
-    proxy_pass            => $proxy_pass_https,
-    proxy_preserve_host   => true,
-    request_headers       => ["set X_FORWARDED_PROTO 'https'"],
+  class { 'foreman::config::apache':
+    passenger     => false,
+    ssl           => true,
+    ssl_cert      => $certs::apache::apache_cert,
+    ssl_key       => $certs::apache::apache_key,
+    ssl_ca        => $certs::apache::ca_cert,
```
Comment (ekohl) [inline manifests/apache.pp]: If we want to mirror the production setup (which supports custom certs), should this also set the chain? I believe this should fix it.
```suggestion
    ssl_ca        => $certs::apache::ca_cert,
    ssl_chain     => $certs::apache::apache_ca_certs,
```

File: manifests/init.pp [hunk 8]
```diff
@@ -150,7 +150,7 @@
 
   if $manage_bundler {
     class { 'katello_devel::setup':
-      require   => Class['katello::candlepin'],
+      require   => [Class['katello::candlepin'], Class['katello::pulp']],
```
Comment (ekohl) [inline manifests/init.pp]: Why is this needed?

File: manifests/init.pp [hunk 9]
```diff
@@ -178,37 +178,11 @@
...
   }
 
-  User<|title == $user|>{groups +> 'qpidd'}
-
-  # TODO: Use katello::pulp
-  class { 'certs::qpid_client':
-    require => Class['pulp::install'],
-    notify  => Class['pulp::service'],
+  class { 'katello::pulp':
+    require => Class['katello_devel::database']
```
Comment (ekohl) [inline manifests/init.pp]: Pulp (2) doesn't use the postgresql database, so why is this needed?

**Comment by ehelms** [comment]
All cleaned up and ready to party

File: manifests/init.pp [hunk 10]
```diff
@@ -178,37 +178,9 @@
     require => Class['katello_devel::database'],
   }
 
-  User<|title == $user|>{groups +> 'qpidd'}
-
-  # TODO: Use katello::pulp
```
Comment (ekohl) [inline manifests/init.pp]: Yay for cleaning up my TODOs
